### PR TITLE
Add graph to Pi-Hole showing DHCP leases if DHCP option is used

### PR DESCRIPTION
**Pull request by Chewie9999**
Pi-Hole does not use the standard DHCP server, so the DHCP agent module in LibreNMS won't work on Pi-Holes.

The agent has been changed to allow 2 more values to be exported - total number of IP addresses in the scope, and how many are actively leased.
The agent has already been changed in PR #351: https://github.com/librenms/librenms-agent/pull/351

This will yield a new graph below the other Pi-Hole Stats in the App screen:
pi-hole-dhcpgraph

If the DHCP function is not used in the Pi-Hole, values will be graphed as "0".

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/polling/applications/pi-hole.inc.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
 $pihole = snmp_walk($device, $oid, $options);
 
 if ($pihole) {
-    [$domains_blocked, $dns_query, $ads_blocked, $ads_percentage, $unique_domains, $queries_forwarded, $queries_cached, $query_a, $query_aaaa, $query_ptr, $query_srv] = explode("\n", $pihole);
+    [$domains_blocked, $dns_query, $ads_blocked, $ads_percentage, $unique_domains, $queries_forwarded, $queries_cached, $query_a, $query_aaaa, $query_ptr, $query_srv,$dhcp_scopesize,$dhcp_leases] = explode("\n", $pihole);
```
Comment (SourceDoctor) [inline includes/polling/applications/pi-hole.inc.php:26]: missing a space before $dhcp_leases

**Review by SourceDoctor — COMMENTED** [review]
could you update Application Documentation about the dependency?

File: includes/polling/applications/pi-hole.inc.php (hunk 1 above)
Comment (Chewie9999) [reply includes/polling/applications/pi-hole.inc.php]: Corrected spacing

**Comment by Jellyfrog** [comment]
@Chewie9999 we're still missing test data for this

**Comment by Jellyfrog** [comment]
I don't have any pi-hole so I can't do it for you but it shouldn't be too tricky.
Install snmpsim 
`pip3 install snmpsim`
Run 
`./scripts/collect-snmp-data.php -h HOSTNAME-of-pihole -v pihole -m applications`
And then run
`./scripts/save-test-data.php --os linux -v pihole -m applications`

**Comment by Chewie9999** [comment]
> `./scripts/save-test-data.php --os linux -v pihole -m applications`

No matter what i do, I can't get that command to work. If I have certain devices already in LibreNMS (linux-based seems to be the pattern), it complains:
`linux_pihole: Cannot add 127.0.0.1, already have device with this IP 127.0.0.1 (`<one of many IPs of devices in librenms)`). You may force add to ignore this.`
If I delete enough of them, that gets rid of the message, but I still get timeouts, even after forcing snmpsim to use 127.0.0.1:1161  and disabling the firewall:

```
[librenms@localhost ~]$ ./scripts/save-test-data.php --os linux -v pihole -m applications
Starting snmpsim listening on 127.0.0.1:1161... 
OS: linux
Module: applications
Variant: pihole

linux_pihole: Could not connect to 127.0.0.1, please check the snmp details and snmp reachability
```
Using `ss` I can see traffic in the receive queue, so the port is open during this time.
What can I try next?

**Comment by PipoCanaja** [comment]
Hi @Chewie9999 
Please add the snmprec file to this PR. I'll run the save-test-data for you if you have issues configuring it. 
Bye

**Comment by Chewie9999** [comment]
Thanks @PipoCanaja:
[linux_pihole.snmprec.txt](https://github.com/librenms/librenms/files/6273146/linux_pihole.snmprec.txt)

**Comment by PipoCanaja** [comment]
@Chewie9999 Could you please commit the file directly to the PR (not in comments). 
Thanx

**Comment by Chewie9999** [comment]
> @Chewie9999 Could you please commit the file directly to the PR (not in comments).
> Thanx

Sorry, now done

**Comment by Jellyfrog** [comment]
First of all, thank you for your contribution!

We're closing this due to inactivity, feel free to reopen if you find the time to continue on this!

**Pull request closed without merging**